corehq/elastic.py

```diff
@@ -205,11 +205,26 @@ def iter_export_docs():
     return ScanResult(query.count(), iter_export_docs())
 
 
-def scroll_query(index_cname, q, es_instance_alias=ES_DEFAULT_INSTANCE):
+def scroll_query(index_cname, query, es_instance_alias=ES_DEFAULT_INSTANCE, **kw):
+    """Perfrom a scrolling search, yielding each doc until the entire context
+    is exhausted.
+
+    :param index_cname: Canonical (registered) name of index to search.
+    :param query: Dict, raw search query.
+    :param es_instance_alias: Name of Elastic instance (for interface instantiation).
+    :param **kw: Additional scroll keyword arguments. Valid options:
+                 `size`: Integer, scroll size (number of documents per "scroll")
+                 `scroll`: String, time value specifying how long the Elastic
+                           cluster should keep the search context alive.
+    """
+    valid_kw = {"size", "scroll"}
+    if not set(kw).issubset(valid_kw):
```

Comment:
I chose to use `**kw` here rather than explicit `size=None, scroll=None` because `None` is a meaningful value to pass.  I first considered defining a `KW_NOTSET = object()` and then using `size=KW_NOTSET, scroll=KW_NOTSET`, but then decided that just validating the `kw` dict was a lot cleaner.  I'm open to other opinions regarding what I've done here.